### PR TITLE
[Documentation] Adding contributing.md

### PR DESCRIPTION
**Pull request by jasonmit**
Lets iterate on this since I'm not entirely familiar with your process.

/cc @ericf

### CODE REVIEW

**Comment by caridy** [comment]
I guess the info in formatjs.io is more about how to contribute to formatjs umbrella of projects, but the contributing.md seems necessary for github to provide meaningful information about how to do the right thing when submitting a PR, how to run tests, etc., I will be ok with a contributing.md per library, as we normally do for all other yahoo pkgs.

**Comment by jasonmit** [comment]
:+1: let me know if the language should change at all or if this is a good enough starting point.  @ericf just expressed interest in needing a contributing.md a few weeks back

**Comment by caridy** [comment]
wording is fine @jasonmit, I think we should add installation and test commands details, that's all from my side.

**Comment by ericf** [comment]
My main concern is just to prompt people to look at the past issues before opening dups.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+5. Follow syntax guidelines detailed below.
+
+6. Push the changes to your fork and submit a pull request.  If this resolves any issues, please mark in the body `resolve #ID` within the body of your pull request.  This allows for github to automatically close the related issue once the pull request is merged.
+
+7. Last step, [submit the pull request][pr]!
+
+[pr]: https://github.com/yahoo/react-intl/compare/
+[fork]: https://github.com/yahoo/react-intl/fork/
+
+## Reporting a Bug
```
Comment (ericf) [inline CONTRIBUTING.md]: I'd move this above the PR section.

**Comment by jasonmit** [comment]
@caridy I'll need to send a separate pull request to allow the tests to be run outside of saucelabs.

**Pull request merged**